src/main/java/seedu/stock/model/SerialNumberSetsBook.java

```diff
@@ -0,0 +1,160 @@
+package seedu.stock.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.collections.ObservableList;
+import seedu.stock.model.stock.AccQuantity;
+import seedu.stock.model.stock.SerialNumberSet;
+import seedu.stock.model.stock.Source;
+import seedu.stock.model.stock.UniqueSerialNumberSetList;
+
+/**
+ * Wraps all data at the SerialNumberSetsBook level.
+ * Duplicates are not allowed (by .isSameSerialNumberSet comparison).
+ */
+public class SerialNumberSetsBook implements ReadOnlySerialNumberSetsBook {
+
+    private final UniqueSerialNumberSetList serialNumberSets;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        serialNumberSets = new UniqueSerialNumberSetList();
+    }
+
+    public SerialNumberSetsBook() {}
+
+    /**
+     * Creates a SerialNumberSetsBook using the SerialNumberSets in the {@code toBeCopied}.
+     */
+    public SerialNumberSetsBook(ReadOnlySerialNumberSetsBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
+
+    /**
+     * Replaces the contents of the SerialNumberSet list with {@code serialNumberSetList}.
+     * {@code serialNumberSets} must not contain duplicate SerialNumberSet.
+     */
+    public void setSerialNumberSets(List<SerialNumberSet> serialNumberSetList) {
+        this.serialNumberSets.setSerialNumberSets(serialNumberSetList);
+    }
+
+    /**
+     * Resets the existing data of this {@code SerialNumberSetsBook} with {@code newData}.
+     */
+    public void resetData(ReadOnlySerialNumberSetsBook newData) {
+        requireNonNull(newData);
+
+        setSerialNumberSets(newData.getSerialNumberSetsList());
+    }
+
+    //// person-level operations
+
+    /**
+     * Returns true if a SerialNumberSet with the same identity as {@code serialNumberSet} exists
+     *     in the SerialNumberSetsBook.
```

Comment:
Is there an extra tab here?

Reply:
I think I read that for javadocs next line should be 4 spaces, let me confirm again.